klog_test.go

```diff
@@ -1921,3 +1921,65 @@ func TestStopFlushDaemon(t *testing.T) {
 		t.Error("expected flushD to be stopped")
 	}
 }
+
+func TestBackTrace(t *testing.T) {
+	const skippedNumLines = 2
+	const sizeNumLines = 1
+	const numLinesPerTrace = 2
+
+	_, callerFile, _, _ := runtime.Caller(0)
+	s1 := fmt.Sprintf("%v", Backtrace())
```

Comment:
```
(dlv) p s1
"k8s.io/klog/v2.Backtrace():\n\t/nvme/gopath/src/k8s.io/klog/klog.go:1661\nk8s.io/klog/v2.TestBackTrace():\n\t/nvme/gopath/src/k8s.io/klog/klog_test.go:1931\ntesting.tRunner():\n\t/nvme/gopath/go-1.18.1/src/testing/testing.go:1439\n"
```

This isn't correct: it includes `Backtrace()`, which should always be skipped. The test doesn't catch that and therefore isn't precise enough.

Instead of checking that that the current function is in the stack trace, it must check that the stack trace starts with it.